### PR TITLE
Added documentation of the new support for Knative deployment

### LINKED ISSUE #250: document use of --knative flag on appsody deploy
We need to document this in the build and deploy document.

The documentation must discuss the following:

If the --knative flag is present and there is a app-deploy.yaml containing an appsodyapplication, the createKnativeService: true tag will be added to the yaml file if not present.

If the --knative flag is not present and createKnativeService: true is in the app-deploy.yaml file the createKnativeService element will be removed. if it is createKnativeService: false, it will remain.

If the stack does not contain an app-deploy.yaml file containing and appsodyapplication, the --knative flag will have no impact.

Unless the appsody deploy is done with a --tag, -t option the imagename will be prefixed with dev.local/

### PR DESCRIPTION
**Pull request by chilanti**
## Checklist

- [x] commit message follows [commit guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).
- [x] website builds and runs in production - for information on how to test locally go [here](https://github.com/appsody/website/blob/master/DEVELOPMENT.md).

## Description
Added verbiage and fixed documentation in light of the new `--knative` flag in `appsody deploy`

## Related Issues
Fixes #250

### CODE REVIEW

**Comment by chilanti** [comment]
@kewegner - please review, but don't merge until we release the corresponding CLI functionality.

File: content/docs/using-appsody/building-and-deploying.md [hunk 1]
```diff
@@ -170,48 +180,20 @@ rules:
...
   Once you have these roles in place, you need to create the appropriate RoleBinding and ClusterRoleBinding to bind your users or groups to them.
 
-
-
-
-
 ### Deployment as a Knative Service
 
-If the stack you are using for your Appsody project does not support the Appsody operator, `appsody deploy` will default to deploying your app as a Knative serving service.
+There are two ways to deploy your app as a Knative service:
```
Comment (neeraj-laad) [inline content/docs/using-appsody/building-and-deploying.md]: Can we reword this to say to deploy an appsody application you can run `appsody deploy --knative` from the project directory. I want users to know that from their point of view that is a single way to deploy to Knative.

Then we can go on and explain that under the covers there are 2 slightly different mechanisms that achieve this, based on the support for the operator in the stack.

Could we also suggest a way for the users to figure out of their stack supports the operator or not?

File: content/docs/using-appsody/building-and-deploying.md [hunk 2]
```diff
@@ -224,24 +206,24 @@ appsody deploy
...
-appsody deploy -t <myaccount/appsody-project> --push --namespace mynamespace
+appsody deploy -t <myaccount/appsody-project> --push --namespace mynamespace [--knative]
 ```
 The command completes the following actions:
 
 - Calls `appsody build` and creates a deployment image, as described in the previous section.
 - The `-t myaccount/appsody-project` option tags the image.
 - The `--push` flag tells the appsody CLI to push the image to Docker Hub.
-- Creates a deployment manifest file named `app-deploy.yaml`, in a `/deploy` subdirectory of the project directory. This yaml file is used to issue a `kubectl apply -f` command against the target Kubernetes cluster. The format of this yaml file depends on whether or not the stack you are using is enabled for the Appsody operator.
+- Creates a deployment manifest file named `app-deploy.yaml`, in a `/config` subdirectory of the project directory. This yaml file is used to issue a `kubectl apply -f` command against the target Kubernetes cluster. The format of this yaml file depends on whether or not the stack you are using is enabled for the Appsody operator.
```
Comment (neeraj-laad) [inline content/docs/using-appsody/building-and-deploying.md]: I think this should be in the / relative to project dir

File: content/docs/using-appsody/building-and-deploying.md (hunk 1 above)
Comment (chilanti) [reply content/docs/using-appsody/building-and-deploying.md]: The only way I can think of is by having them run `appsody deploy --generate-only` and then inspect the manifest. If it's an `AppsodyApplication`, that means that the stack supports the operator. Other than that... they'd have to look at the structure of the stack. Again - this is another case where we could use some metadata about the stacks.

Comment (chilanti) [reply content/docs/using-appsody/building-and-deploying.md]: @neeraj-laad  Addressed your comments... take a look!
